core/src/main/java/tc/oc/pgm/action/actions/EnchantItemAction.java

```diff
@@ -0,0 +1,45 @@
+package tc.oc.pgm.action.actions;
+
+import java.util.Objects;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.inventory.PlayerInventory;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.inventory.ItemMatcher;
+import tc.oc.pgm.util.math.Formula;
+
+public class EnchantItemAction extends AbstractAction<MatchPlayer> {
+
+  private final ItemMatcher matcher;
+  private final Enchantment enchant;
+  private final Formula<MatchPlayer> level;
+
+  public EnchantItemAction(ItemMatcher matcher, Enchantment enchant, Formula<MatchPlayer> level) {
+    super(MatchPlayer.class);
+    this.matcher = matcher;
+    this.enchant = enchant;
+    this.level = level;
+  }
+
+  @Override
+  public void trigger(MatchPlayer player) {
+    PlayerInventory inv = Objects.requireNonNull(player.getInventory());
```

Comment:
nit: The `Objects.requireNonNull` seems kind of pointless given it will throw a null pointer anyways on line 30 if its null.

Reply:
it is pointless, its just to make the IDE not complain about it being potentially nullable: it's not